### PR TITLE
[Feature] temporary table(part-4): support system table of temporary tables

### PR DESCRIPTION
**Pull request by silverbullet233**
## What I'm doing:

the fifth part of work split from https://github.com/StarRocks/starrocks/pull/43048, this is also the last one.

Add a new system table `information_schema.temp_tables` to query all temporary table informations in the cluster. 
The table structure is similar to `information_schema.tables`, 
on this basis, additional tow columns are added: `SESSION` and `TABLE_ID`

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by packy92 — DISMISSED** [review]

File: be/src/exec/schema_scanner.cpp [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 #include "exec/schema_scanner.h"
 
+#include <protocol/TDebugProtocol.h>
```
Comment (kangkaisen) [inline be/src/exec/schema_scanner.cpp:17]: unused?

File: fe/fe-core/src/main/java/com/starrocks/service/InformationSchemaDataSource.java [hunk 2]
```diff
@@ -465,6 +471,80 @@ public static TGetTablesInfoResponse generateTablesInfoResponse(TGetTablesInfoRe
...
+        for (String dbName : result.authorizedDbs) {
+            Database db = metadataMgr.getDb(catalogName, dbName);
+            if (db != null) {
+                requiredDbIds.add(db.getId());
+            }
+        }
+
+        com.google.common.collect.Table<Long, UUID, Long> allTables = temporaryTableMgr.getAllTemporaryTables(requiredDbIds);
+
+        List<TTableInfo> tableInfos = new ArrayList<>();
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/service/InformationSchemaDataSource.java:497]: you should handle limit.

File: be/src/exec/schema_scanner.cpp (hunk 1 above)
Comment (silverbullet233) [reply be/src/exec/schema_scanner.cpp]: fixed

File: fe/fe-core/src/main/java/com/starrocks/service/InformationSchemaDataSource.java (hunk 2 above)
Comment (silverbullet233) [reply fe/fe-core/src/main/java/com/starrocks/service/InformationSchemaDataSource.java:497]: fixed

**Review by wangsimo0 — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by LiShuMing — APPROVED** [review]

**Comment by silverbullet233** [comment]
@mergify backport branch-3.3

**Pull request merged**